scalyr_agent/builtin_monitors/url_monitor.py

```diff
@@ -103,11 +107,62 @@ def _initialize(self):
         else:
             self.extractor = None
 
-    def gather_sample(self):
+    @staticmethod
+    def form_request(url, request_method, request_data, headers, logger):
+        """
+        Forms the HTTP request based on the request URL, HTTP headers and method
+        @param url: HTTP request URL
+        @param request_method: HTTP request method
+        @param request_data: HTTP request payload data
+        @param headers: HTTP header
+        @param logger: The logger to use
+
+        @type url: str
+        @type request_method: str
+        @type request_data: str | None
+        @type headers: JsonArray (of tuples with header name, header value combination)
+        @type logger: logging.Logger or subtypes
+        @return: Request object
+        """
+
+        request = urllib2.Request(url, data=request_data)
+        if headers:
+            if type(headers) != JsonArray:
+                logger.emit_value(
+                    'URL Monitor has malformed optional headers: {}'.format(repr(headers))
+                )
+            else:
+                for header in headers:
+                    request.add_header(header["header"], header["value"])
+
+        # seems awkward to override the GET method, but internally it flips
+        # between GET and POST anyway based on the existence of request body
+        request.get_method = lambda: request_method
+        return request
+
+    def gather_sample(self, request_method='GET', request_data=None, headers=None):
```

Comment:
`gather_sample` is defined in the base class, so overriding it and changing what arguments it takes seems like it's not a good idea.  The `request_method`, `request_data`, `headers` are instance variables, why not just get them from `self`?